rose/server/web/settings/settings-form.js

```diff
@@ -0,0 +1,82 @@
+class SettingsForm extends HTMLElement {
+    constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+    }
+
+    connectedCallback() {
+        this.render();
+        this.loadDrivers();
+        this.shadowRoot.querySelector('#submit-button').addEventListener('click', this.sendDrivers.bind(this));
+        this.shadowRoot.querySelector('#cancel-button').addEventListener('click', this.cancelSetting.bind(this));
+    }
+
+    render() {
+        this.shadowRoot.innerHTML = `
```

Comment:
Interesting take. I would have thought to not use sha soon soon in this case and just this.querySelector, but this is good too

You may choose to use a template element is a static init block, but this isn't reusable so the gains are few

If you'll stick with shadow DOM, use a constructible style sheet and adopted stylesheets instead of a link element

Reply:
@bennypowers that was a nice learning experience :-)
not using template, but separated the template and css from the main js component, WDYT ?